README.md

```diff
@@ -251,6 +262,77 @@ The shared storage worklet invocation methods (`addModule`, `createWorklet`, and
 *  The response header will only be honored if the corresponding request included the request header: `Sec-Shared-Storage-Writable: ?1`.
 *  See example usage below.
 
+### Locking for Modifier Methods
+
+All modifier methods (`set`, `append`, `delete`, `clear`), whether invoked from JavaScript or from response headers, accept a `withLockOnResource: <resource>` option. This option instructs the method to acquire a lock on the designated resource before executing.
+
+The locks requested this way are partitioned by the shared storage data origin, and are independent of any locks obtained via `navigator.locks.request` in a Window or Worker context. Note that they share the same scope with the locks obtained via `navigator.locks.request` in the SharedStorageWorklet context.
+
+Unlike `navigator.locks.request`, which offers various configuration options, the locks requested this way always use the default settings:
+*  `mode: "exclusive"`: The lock is never shared with other locks.
+*  `steal: false`: The lock will not preempt other locks.
+*  `ifAvailable: false`: If the lock is currently held by others, keep waiting and don't skip.
+
+#### Example: Naive Integrity Check
```

Comment:
I see the race here, but it's pretty unclear what the purpose is.

Perhaps instead, write to three variables, a, b, and c from the document and within the worklet read all three and add a comment that if some document (which could be in another tab!) is writing while the worklet is reading, then they may not be consistent.

Reply:
As I mentioned in [a previous thread](https://github.com/WICG/shared-storage/pull/197#discussion_r1767828720), I'm finding it difficult to write a compelling example for this. Two concerns:

1. Regarding your three-variable example, what specific use case does it illustrate? (Reach? A/B?) What do the variables represent? We can certainly leave it vague.
2. If consistency is crucial, I suspect the use case will encounter [Caveat 2: No atomicity for multiple writes](https://github.com/WICG/shared-storage/blob/yao-web-locks/README.md#caveat-2-no-atomicity-for-multiple-writes). This could make the example less effective.

Ultimately, I believe this locking mechanism will likely be ineffective without a "multi-write" operation. PLMK if you still prefer updating the example. Thanks!